### DISCUSSION TITLE
start application on worspace (only when first starting)

### DISCUSSION THREAD

**Discussion by yuri0r** [i3/i3#4524] [Q&A] [answered]
so according to the [documentation](https://i3wm.org/docs/userguide.html#assign_workspace) i should be able to have a terminal start on workspace 2 when I first start into i3 without having to assign all my terminals to workspace 2.
in my i3 config i have:

`exec --no-startup-id i3-msg 'workspace 2; exec i3-sensible-terminal; workspace 1'`

but the terminal (st) does start on workspace 1 when I log into i3. am I doing something wrong? how should I go about setting up an initial layout when I first log in but it's impractical to assign the programs as they aren't meant to permanently spawn on the same workspace?

**Comment by Airblader** [comment, marked as answer]
This relies on the application supporting the startup notifications spec. I don't know if st does or not, that's something you'd have to check.

An alternative would be to use layout [saving and restoring](https://i3wm.org/docs/layout-saving.html).

**Reply by yuri0r** [reply]
sure enough, there is no explicit statement from suckless I found but urxvt worked perfectly.
